internal/host/host.go

```diff
@@ -1441,14 +1441,28 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 }
 
 func (m *Manager) UnRegisterHost(ctx context.Context, h *models.Host) error {
-	if err := common.DeleteHostFromDB(m.db, h.ID.String(), h.InfraEnvID.String()); err != nil {
+	if h.ClusterID == nil {
+		return fmt.Errorf("UnRegisterHost failed to locate ClusterID")
+	}
+	var cluster common.Cluster
+	err := m.db.Select("status").Take(&cluster, "id = ?", h.ClusterID.String()).Error
+	if err != nil {
 		return err
 	}
-	if h.ClusterID != nil {
-		if err := m.db.Model(&common.Cluster{}).Where("id = ?", h.ClusterID).Update("trigger_monitor_timestamp", time.Now()).Error; err != nil {
-			return err
+	clusterProgressingStates := []string{models.ClusterStatusInstalling, models.ClusterStatusFinalizing}
```

Comment:
I think we should block it also in case of preparing-for-installation, @avishayt WDYT?

Reply:
Yes, and also installing-pending-user-action.  I suggest putting this list of states into internal/cluster/cluster.go, similar to CanDownloadFiles() or CanDownloadKubeconfig().

Another thing, while looking at that file, I see three places where we check for ClusterStatusInstalling but ignore ClusterStatusInstallingPendingUserAction:
1. CancelInstallation() doesn't allow canceling while in installing-pending-user-action
2. IsOperatorAvailable() will return false while in installing-pending-user-action
3. SetVipsData() won't catch VIPs changing during installation

Reply:
@avishayt so we can't move it to cluster package due to import loop issues
so I've create it in common

let me know if it's ok i will start working on the test's